includes/Emails/VendorNewOrder.php

```diff
@@ -0,0 +1,193 @@
+        $this->template_html  = 'emails/vendor-new-order.php';
+        $this->template_plain = 'emails/plain/vendor-new-order.php';
+        $this->template_base  = DOKAN_DIR.'/templates/';
+        $this->placeholders   = array(
+            '{site_title}'   => $this->get_blogname(),
+            '{order_date}'   => '',
+            '{order_number}' => '',
+        );
+
+        // Triggers for this email.
+        add_action( 'woocommerce_order_status_pending_to_processing_notification', array( $this, 'custom_trigger_email_action' ), 10, 2 );
+        add_action( 'woocommerce_order_status_pending_to_completed_notification', array( $this, 'trigger' ), 10, 2 );
+        add_action( 'woocommerce_order_status_pending_to_on-hold_notification', array( $this, 'trigger' ), 10, 2 );
+        add_action( 'woocommerce_order_status_failed_to_processing_notification', array( $this, 'trigger' ), 10, 2 );
+        add_action( 'woocommerce_order_status_failed_to_completed_notification', array( $this, 'trigger' ), 10, 2 );
+        add_action( 'woocommerce_order_status_failed_to_on-hold_notification', array( $this, 'trigger' ), 10, 2 );
+        
+        // Call parent constructor.
+        parent::__construct();
+        
+        // Other settings.
+        $this->recipient = 'vendor@ofthe.product';
+    }
+
+    /**
+     * Get email subject.
+     *
+     * @since  3.1.0
+     * @return string
+     */
+    public function get_default_subject() {
+        return __( '[{site_title}] New customer order ({order_number}) - {order_date}', 'woocommerce' );
+    }
+
+    /**
+     * Get email heading.
+     *
+     * @since  3.1.0
+     * @return string
+     */
+    public function get_default_heading() {
+        return __( 'New Customer Order: #{order_number}', 'woocommerce' );
+    }
+
+    /**
+     * Trigger the sending of this email.
+     *
+     * @param int            $order_id The order ID.
+     * @param WC_Order|false $order Order object.
+     */
+    public function custom_trigger_email_action( $order_id, $order = false ) {
+
+        if ( ! $this->is_enabled() ) {
+            return;
+        }
+
+        $this->setup_locale();
+        if ( $order_id && ! is_a( $order, 'WC_Order' ) ) {
+            $order = wc_get_order( $order_id );
+        }
+
+        if ( is_a( $order, 'WC_Order' ) ) {
+            $this->object                           = $order;
+            $this->placeholders['{order_date}']     = wc_format_datetime( $this->object->get_date_created() );
+            $this->placeholders['{order_number}']   = $this->object->get_order_number();
+        }
+        
+        $sellers 	  = dokan_get_seller_id_by_order( $order_id );
+        if ( empty( $sellers ) ) {
+            return;
+        }
+        
+        // check has sub order 
+        if ( get_post_meta( $order_id, 'has_sub_order', true ) ) {
```

Comment:
Can be used `dokan_is_sub_order` function to check whether the order is sub order or not.